modules/dbfs/filestorage.go

```diff
@@ -76,7 +79,110 @@ func (di *DatabaseImpl) FileMove(startRelpath string, startFilename string, endR
 	return err
 }
 
-func calculateFilePath(relpath string, filename string, projectID int64) (string, error) {
+// returns the swap file contents and any error
+func (di *DatabaseImpl) makeSwp(relpath string, filename string, projectID int64) ([]byte, error) {
+	relFilePath, err := di.calculateFilePath(relpath, filename, projectID)
+	if err != nil {
+		return []byte{}, err
+	}
+	fileLocation := filepath.Join(relFilePath, filename)
+	swapLoc := di.getSwpLocation(fileLocation)
+
+	err = di.fileCopy(fileLocation, swapLoc)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	fileBytes, err := ioutil.ReadFile(swapLoc)
+	return fileBytes, err
+}
+
+// swapRead returns the swap file from the calculated location on the disk
+func (di *DatabaseImpl) swapRead(relpath string, filename string, projectID int64) (*[]byte, error) {
+	relFilePath, err := di.calculateFilePath(relpath, filename, projectID)
+	if err != nil {
+		return new([]byte), err
+	}
+	fileLocation := filepath.Join(relFilePath, filename)
+	swapLocation := di.getSwpLocation(fileLocation)
+	fileBytes, err := ioutil.ReadFile(swapLocation)
+	return &fileBytes, err
+}
+
+// FileWriteToSwap writes the swapfile for the file with the given info
+func (di *DatabaseImpl) FileWriteToSwap(meta FileMeta, raw []byte) error {
```

Comment:
In go, capitalized method names mean that the function is exported from the package. This function is needed inside of the scrunching package, so it needs to be exported from the dbfs package. Think of it like the public keyword in Java